### PR TITLE
Add useful components and fix .gitignore

### PR DESCRIPTION
**Pull request by IsakSundeSingh**
Adds some triggers, converters and behaviors.

### CODE REVIEW

File: Diplomatic/Utils/Converters/PositiveLength.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using System.Globalization;
+using Xamarin.Forms;
+
+namespace Diplomatic.Utils.Converters
+{
+    public class PositiveLength : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            return (int)value > 0;
```
Comment (fredeil) [inline Diplomatic/Utils/Converters/PositiveLength.cs]: Will crash, consider using C# 7.0 pattern matching instead

File: Diplomatic/Utils/Converters/RegexMatch.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public class RegexMatch : IValueConverter
+    {
+        public string Pattern { get; set; } = "";
+
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            string text = (string)value;
+            Regex regex = new Regex(Pattern);
+            return regex.IsMatch(text);
```
Comment (fredeil) [inline Diplomatic/Utils/Converters/RegexMatch.cs]: You should replace this inefficient code with a call to the static [Regex.IsMatch(String, String)](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch#System_Text_RegularExpressions_Regex_IsMatch_System_String_System_String_) method.

**Comment by fredeil** [comment]
Should add an .editorconfig to the project too.
https://github.com/Dualog-students/dualog-students.github.io/blob/master/.editorconfig

**Comment by MaxJohansen** [comment]
I suggest the following additions to the root element of the editorconfig:

```
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
```

Because trailing whitespace is gross and final newlines are POSIX standard.

**Review by MaxJohansen — CHANGES_REQUESTED** [review]
I suggest the following additions to the root element of the editorconfig:

```
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
```

Because trailing whitespace is gross and final newlines are POSIX standard.

**Review by vebjornhaugland — APPROVED** [review]
Looks good

**Comment by IsakSundeSingh** [comment]
Let's 🚢 🇮🇹

**Review by MaxJohansen — APPROVED** [review]

**Pull request merged**